README.md

```diff
@@ -19,3 +19,115 @@ The key stakeholder for this application is the Associate Department Head, who i
     	- Sprint 2:
        		- Goal: Have a raw views for all the data upload.
          	- (Sprint Plan)[https://github.com/tamu-edu-students/Faculty-Teaching-Assignment/blob/main/documentation/Fall2024/Sprint_2_Plan.pdf]
+
+# Getting Started with Setup and Deployment
+
+## Prerequisites
+1. **Heroku Account**: [Sign up for a Heroku account](https://signup.heroku.com/).
+2. **Heroku CLI**: Download and install the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli).
+3. **GitHub Repository**: Ensure your Rails application is committed and pushed to a GitHub repository.
+4. **Rails and Ruby**: Check that you have Ruby and Rails installed locally to set up the app.
+5. **OAuth Configuration**: Ensure you have Google OAuth API keys
+
+---
+
+## 1. Clone the GitHub Repository (If Needed)
+If you're setting up the project locally, clone the repository:
+
+```bash
+git clone https://github.com/tamu-edu-students/Faculty-Teaching-Assignment.git
+cd Faculty-Teaching-Assignment
+```
+
+---
+
+## 2. Install Dependencies and setup Rails Master Key
+Run `bundle install` to install all required dependencies.
+
+```bash
+bundle install
+```
+Generate your ```GOOGLE_CLIENT_ID``` and ```GOOGLE_CLIENT_SECRET```. Encrypt both of these using a ```RAILS_MASTER_KEY``` and store it in your ```config/credentials.ymc.enc``` 
```

Comment:
This is done on the Google Developer console, not through Rails. This [link](https://github.com/tamu-edu-students/Google-Auth-Ruby-By-JD#setup-google-oauth-on-googles-end) provides details on how to generate them